### DISCUSSION TITLE
Does it make sense to use a BaseModel is the basis for a custom class in a standalone package?

### DISCUSSION THREAD

**Discussion by FISHMANPET** [pydantic/pydantic#5666] [Question]
I'm sure there's a better way to phrase the title, but here's my situation.

I'm working on a [package](https://github.com/umn-microsoft-automation/grouper-python) that creates a Pythonic wrapper around an external REST API.  I've created a basic "Client" class, and then you use that Client to retrieve objects from the API (in this case, Group, Stem, Subject, Person).  Each of those objects has methods that run against the instance.  For example, I can get a Group object that represents a specific group in Grouper, then I can call `group.get_members()` to get the members of that group.

Right now, as you can see, those objects are all subclasses of Pydantic's BaseModel.  It all obviously _works_ but I'm wondering what the community thinks of this approach in general.  I gravitated towards Pydantic after having lots of experience with it via FastAPI, but at this point, I'm not sure it's getting me any benefit, and I'm worried that I'm going to discover somewhere along the way that it ends up breaking something.

One of the examples where Pydantic hasn't been helpful is in actually constructing the objects.  I've ended up writing a `from_result()` class method that will construct the object based on a payload from the API, rather than constructing new objects by passing in each parameter.  I've attached a "Client" object to each of these, which Pydantic can't validate so I've had to add `arbitrary_types_allowed` to the Config.  I don't find myself really taking advantage of any of the data validation features of Pydantic right now.

If someone has a better way of phrasing the initial question I can change the title, but I couldn't really find any similar discussion like this here.

**Comment by chrisK824** [comment]
Not sure what are you trying to achieve here. Can you provide an example on what did you expect to use pydantic basemodel for and what you are doing instead?

**Reply by FISHMANPET** [reply]
To be honest I'm not sure what I was expecting.  To be clear, everything _works_, I don't have any errors when using Pydantic like this.  But I also spent a few minutes and replaced all my classes with either classes that don't inherit from anything, or the built-in dataclass in a few cases, and all my code still worked just fine.

This is more of a philosophical question than a technical one.

As I've been developing this, I've enjoyed having Pydantic's built-in methods like __str__, __repr__, dict(), json(), etc.  Without Pydantic I'd have to write much of that myself.  I imagine for someone using my library they would enjoy some of that functionality as well, so either I use Pydantic or write it myself.

I went back and listened to some of the Podcasts linked on the home page, hoping listening to Samuel might give me a better idea of what's appropriate and what's not, and it didn't really provide much clarity one way or the other, as he says that lots of people are doing lots of things with Pydantic beyond what he initially intended.  Maybe if I don't particularly care about the validation features of Pydantic, I can still use it because of some of the data presentation features it offers.

The only other dependency for my package is `httpx`, and maybe another concern is that if I'm introducing a dependency like pydantic that it's "worth it" to anybody consuming my package.
